### PR TITLE
feat(dialog): adding component

### PR DESCRIPTION
**Pull request by inavac182**
## 🔥 Adding dialog component
----------------------------------------------

### Description
This is the dialog component

#### Before
N/A

#### After
Apr-20-2023 22-35-02

### CODE REVIEW

File: src/providers/dialogs-controller.ts [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+    },
+    [openedDialogs, setOpenedDialogs]
+  );
+
+  return {
+    openDialog,
+    closeDialog,
+    openedDialogs,
+  };
+};
```
Comment (inavac182) [inline src/providers/dialogs-controller.ts:32]: This is the controller that keeps the dialogs state inside the docs page, each client will have to create something similar to this and can do whatever they want to do for handling the dialogs state, in this case I'm using a useState to keep track of the opened dialogs but in a different client they could use the URL for instance or whatever other thing that fit their requirements.

File: packages/dialog/src/provider/dialogs-provider.ts [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+  console.error('No dialog controller implemented');
+};
+
+export const UiDialogsControllerContext = React.createContext<IDialogController>({
+  openDialog: noOpDialogControl,
+  closeDialog: noOpDialogControl,
+  openedDialogs: [],
+});
+
+UiDialogsControllerContext.displayName = 'UiDialogsControllerContext';
```
Comment (inavac182) [inline packages/dialog/src/provider/dialogs-provider.ts:19]: This is the context that will be used for each client to provide a state handling, we could provide a very generic handling from the repo, but as of now its a noOp, which means client have to implement it.

File: packages/dialog/src/provider/use-dialog-hook.ts [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+      if (dialog === dialogId) {
+        opened = true;
+      }
+    });
+
+    setIsOpen(opened);
+  });
+
+  return { isOpen, actions };
+};
```
Comment (inavac182) [inline packages/dialog/src/provider/use-dialog-hook.ts:38]: This is the hook that will be used inside each component and will return the functions that each instance of a dialog can use to open or close.

**Review by saudelog — APPROVED** [review]

File: packages/dialog/__tests__/ui-dialog.spec.tsx [hunk 4]
```diff
@@ -0,0 +1,134 @@
...
+    render(<MockedNotCorrectlySetupComponent />);
+
+    expect(screen.queryByText('Dialog content')).not.toBeInTheDocument();
+    fireEvent.click(screen.getByText('Open Dialog'));
+
+    expect(screen.queryByRole('dialog')).not.toBeInTheDocument();
+    expect(console.error).toHaveBeenCalledTimes(1);
+    expect(console.error).toHaveBeenCalledWith('No dialog controller implemented');
+
+    console.error = consoleError;
```
[the remark this answers is not in the document]
Comment (inavac182) [reply packages/dialog/__tests__/ui-dialog.spec.tsx:132]: This is for the unit tests mocking, as a `console.error` triggers an error in jest, I mocked it so I can verify that we trigger that error from our code without breaking the unit test, once I verify it then I put back to the previous value.

**Pull request merged**